org-transclusion.el

```diff
@@ -562,6 +568,35 @@ the rest of the buffer unchanged."
       (move-marker marker nil) ; point nowhere for GC
       list)))
 
+(defun org-transclusion-toggle ()
+  "Toggle transclusion in the current line."
+  (interactive)
+  (condition-case nil
+      (org-transclusion-add)
+    (error (condition-case nil
+               (org-transclusion-remove)
+             (error (progn (org-transclusion-prefix-insert)
+                           (org-transclusion-add))))))
+  (if org-transclusion-latex-preview-on-toggle
+      (org-latex-preview '(16))))
+
+(defun org-transclusion-toggle-all ()
+  "Toggle all transclusions in the current buffer."
+  (interactive)
+  (condition-case nil
+      (progn
+        (org-transclusion-remove)
+        (org-transclusion-remove-all))
+    (error (org-transclusion-add-all)))
+  (if org-transclusion-latex-preview-on-toggle
+      (org-latex-preview '(16))))
+
+(defun org-transclusion-prefix-insert ()
```

Comment:
`prefix-insert` seems to have the same intention as `org-transclusion-make-from-link` does -- create a `#+transclude` link out of a normal link. 

What would it do if the link is not in the beginning of a paragraph? For instance, it's common to have a paragraph like this:

> His work has had a considerable influence on mathematics, logic, set theory, [[file:linguistics.org][linguistics]], artificial intelligence, cognitive science, computer science, and various areas of analytic philosophy, especially philosophy of mathematics, philosophy of language, epistemology and metaphysics

Reply:
I wasn't aware of that function, thanks for referencing it.

Reply:
Actually, this may be specific to my own purposes, but as you may have seen in my #120 screenshot, I have links by themselves on a line. I essentially want to be able to preview and unpreview things, without necessarily having `#+transclude:` everywhere. `prefix-insert` essentially is part of `org-transclusion-toggle` in allowing links to stay as they are (without `#+transclude:`), but allow preview and unpreview of links.

The analogy is I have Luhmann's slipbox, with each link representing the "title" of the notecards viewable from above, but you can "scroll through" the slipbox by previewing each entry.

Compared to this, `org-transclusion-make-from-link` actually just makes a new link, which seems to be for a different use case.

Reply:
With regard to the example you referenced, I had thought about that and how that could play with `org-transclusion-toggle` in general. I think, again for my own purposes, it would be nice to have something like `org-transclusion-make-from-link` below the paragraph when `org-transclusion-toggle` is called, as `org-transclusion-toggle` for lines with text included doesn't work at the moment.

Reply:
> again for my own purposes, it would be nice to have something like org-transclusion-make-from-link below the paragraph when org-transclusion-toggle is called, as org-transclusion-toggle for lines with text included doesn't work at the moment.

I think you would need to decide whether or not you would like the preview feature to be for "your own purpose" or something to be shared by others.